service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/SenderLinkHandler.java

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.service.transport.amqps;
+
+import com.microsoft.azure.sdk.iot.service.transport.TransportUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.Target;
+import org.apache.qpid.proton.amqp.transport.SenderSettleMode;
+import org.apache.qpid.proton.engine.BaseHandler;
+import org.apache.qpid.proton.engine.Delivery;
+import org.apache.qpid.proton.engine.EndpointState;
+import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.Link;
+import org.apache.qpid.proton.engine.Sender;
+import org.apache.qpid.proton.message.impl.MessageImpl;
+import org.apache.qpid.proton.reactor.FlowController;
+
+import java.io.IOException;
+import java.nio.BufferOverflowException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public abstract class SenderLinkHandler extends BaseHandler
+{
+    private static final String API_VERSION_KEY = "com.microsoft:api-version";
+    Map<Symbol, Object> amqpProperties;
+    String senderLinkTag;
+    String linkCorrelationId;
+    String senderLinkAddress;
+    Sender senderLink;
+    private long nextTag = 0;
+
+    protected final LinkStateCallback linkStateCallback;
+
+    protected abstract String getLinkInstanceType();
+
+    SenderLinkHandler(Sender sender, String linkCorrelationId, LinkStateCallback linkStateCallback)
+    {
+        this.amqpProperties = new HashMap<>();
+        this.amqpProperties.put(Symbol.getSymbol(API_VERSION_KEY), TransportUtils.IOTHUB_API_VERSION);
```

Comment:
Can setting this property be moved to the base constructor as it is comment for all handlers?

Reply:
This is the base constructor. This class is the most generic sender link class that we own, as it only extends the proton-j BaseHandler type